File: leetcode/1365-how-many-numbers-are-smaller-than-the-current-number/solution.cpp
vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
    vector<int> count_lower(101, 0);
    vector<int> result(nums.size(), 0);

    for (int i = 0; i < nums.size(); i++) {
        for (int j = nums[i] + 1; j < 101; j++) {
            count_lower[j]++;
        }
    }

    for (int i = 0; i < nums.size(); i++) {
        result[i] = count_lower[nums[i]];
    }

    return result;
}

File: leetcode/1365-how-many-numbers-are-smaller-than-the-current-number/solution.py
def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
    count_lower = [0] * 101
    result = [0] * len(nums)

    for num in nums:
        j = num + 1
        while (j < 101):
            count_lower[j] += 1
            j += 1

    for i in range(len(nums)):
        result[i] = count_lower[nums[i]]

    return result
